.github/workflows/test.yml

```diff
@@ -77,32 +77,34 @@ jobs:
         if [ -f ${CI_IMAGE_CACHE}${CI_IMAGE_MASTER_TAR}.gz ]; then
             cp ${CI_IMAGE_CACHE}/${CI_IMAGE_MASTER_TAR}.gz ${CI_IMAGE_MASTER_TAR}.gz
             gunzip ${CI_IMAGE_MASTER_TAR}.gz
-            echo "::set-output name=MASTER_IMAGE_RESTORED_FROM_CACHE::true"
+            echo "MASTER_IMAGE_RESTORED_FROM_CACHE=true" >> "$GITHUB_OUTPUT"
+            echo "MASTER_IMAGE_RESTORED=true" >> "$GITHUB_OUTPUT"
             exit 0
         fi
 
         if docker pull ghcr.io/ovn-org/ovn-kubernetes/ovn-kube-f:master; then
             docker tag ghcr.io/ovn-org/ovn-kubernetes/ovn-kube-f:master ovn-daemonset-f:dev
 
-            echo "::set-output name=MASTER_IMAGE_RESTORED_FROM_GHCR::true"
+            echo "MASTER_IMAGE_RESTORED=true" >> "$GITHUB_OUTPUT"
             exit 0
         fi
     # only run the following steps if the master image was not found in the cache
+    - name: Check out code into the Go module directory - from master branch
```

Comment:
why are we checking out the ovn repo before setting up go? any reason behind changing the order here?

Reply:
setup-go@v5 also wants to cache all the go dependencies by default.  But it is looking for go.sum file and it doesn't exist if the repo is not checked out.  That triggers a warning.  So, re-ordering to avoid that.
(It also looks for go.sum in the top-level directory, so we need to supply the paths as an option, otherwise it warns as well.)